pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -63,21 +96,52 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	if !current.CanTransitionTo(next) {
 		return fmt.Errorf("invalid state transition: %s -> %s", current.Name(), next.Name())
 	}
-	// TODO: call pre-transition listeners -  Issue: https://github.com/hyperledger/aries-framework-go/issues/140
-	followup, err := next.Execute(msg)
+
+	// trigger the event if the message type is registered (if it's not outbound)
+	eventsTriggered := false
+	if !msg.Outbound {
+		eventsTriggered, err = s.sendEvent(msg, thid, next)
+		if err != nil {
+			return fmt.Errorf("send events failed: %w", err)
+		}
+	}
+
+	// if no events are trigger continue the execution
+	if !eventsTriggered {
+		return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
```

Comment:
We have to be careful here.

There is a race between starting the protocol and disabling auto execution (currently via register). We don't want to accidentally auto execute state and we don't want to accidentally drop messages either :).

Reply:
valid point. Created #261 .

Reply:
This has been addressed in this PR. Godoc - https://github.com/hyperledger/aries-framework-go/pull/240/files#diff-f761c91e753eba1c42a68076df06ce12R133.